addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -324,19 +326,27 @@ func (r *MirrorPeerReconciler) enableMulticlusterService(ctx context.Context, st
 		Namespace: "submariner-operator"},
 		&submariner)
 	if err != nil {
+		klog.Errorf("Error fetching Submariner config while enabling MCS. Error: %v", err)
 		return err
 	}
 
 	if sc.Spec.Network == nil {
 		sc.Spec.Network = &rookv1.NetworkSpec{}
+		klog.Infof("StorageCluster %q in %q namespace has no network config defined. Initializing it now. New NetworkSpec: %v", storageClusterName, namespace, sc.Spec.Network)
 	}
 
 	if !sc.Spec.Network.MultiClusterService.Enabled || sc.Spec.Network.MultiClusterService.ClusterID == "" {
 		sc.Spec.Network.MultiClusterService.Enabled = true
 		sc.Spec.Network.MultiClusterService.ClusterID = submariner.Spec.ClusterID
-		return r.SpokeClient.Update(ctx, &sc)
+		klog.Infof("StorageCluster %q in %q namespace has MCS disabled. Enabling it now. New MCS spec: %v", storageClusterName, namespace, sc.Spec.Network.MultiClusterService)
+		err := r.SpokeClient.Update(ctx, &sc)
+		if err != nil {
+			klog.Errorf("Error updating MCS config for StorageCluster %q in %q namespace. Error: %v", storageClusterName, namespace, err)
```

Comment:
```suggestion
			klog.Errorf("Error updating MCS config for StorageCluster %q in %q namespace. Error: %v", storageClusterName, namespace)
```
nit : since we are returning it , we don't have to log it here

Reply:
Looked like returned errors are not logged for some reason. So I added it everywhere before returning.